File: project/backend/src/main/java/com/example/faculdade_progressus/models/Aluno.java
package com.example.faculdade_progressus.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@Entity
public class Aluno {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    private String cpf;

    private String email;

    @JsonFormat(pattern = "dd/MM/yyyy")
    private LocalDate dataNascimento;

    @OneToMany(mappedBy = "aluno", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties("aluno")
    private Set<Matricula> matriculas;
}

File: project/frontend/lib/models/aluno.dart
// lib/models/aluno.dart
class Aluno {
  final int id;
  final String nome;
  final String cpf;
  final String email;
  final String dataNascimento;

  Aluno({required this.id, required this.nome, required this.cpf, required this.email, required this.dataNascimento});

  factory Aluno.fromJson(Map<String, dynamic> json) {
    return Aluno(
      id: json['id'],
      nome: json['nome'],
      cpf: json['cpf'],
      email: json['email'],
      dataNascimento: json['dataNascimento'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'cpf': cpf,
      'email': email,
      'dataNascimento': dataNascimento,
    };
  }
}
